File: backend/app/models/company.py
from sqlalchemy import Column, Integer, String, Text, Boolean, ForeignKey, DateTime, func
from sqlalchemy.orm import relationship

from ..database import Base


class Company(Base):
    __tablename__ = "companies"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(200), nullable=False)
    description = Column(Text, nullable=True)
    address = Column(String(255), nullable=True)
    phone = Column(String(50), nullable=True)
    email = Column(String(150), nullable=True)
    website = Column(String(255), nullable=True)
    rccm_number = Column(String(100), nullable=True)
    tax_number = Column(String(100), nullable=True)
    sector = Column(String(120), nullable=True)
    is_verified = Column(Boolean, default=False, nullable=False)
    owner_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    owner = relationship("User", back_populates="entreprise_profile")
    offres = relationship("Offer", back_populates="company")
    marches_gagnes = relationship("PublicContract", back_populates="company")

File: frontend/src/types/company.ts
export interface Company {
  id: number;
  name: string;
  description?: string | null;
  address?: string | null;
  phone?: string | null;
  email?: string | null;
  website?: string | null;
  rccm_number?: string | null;
  tax_number?: string | null;
  sector?: string | null;
  is_verified: boolean;
  owner_id?: number | null;
  created_at: string;
  updated_at: string;
}

export interface CompanyCreate {
  name: string;
  description?: string;
  address?: string;
  phone?: string;
  email?: string;
  website?: string;
  rccm_number?: string;
  tax_number?: string;
  sector?: string;
  owner_id?: number;
}

export type CompanyUpdate = Partial<Omit<CompanyCreate, "owner_id">>;
